### ISSUE TITLE
Base the qwerty-fr on the `UK ISO (standard)` instead of the currently used `US ISO/ANSI (terminal style)`?

### ISSUE THREAD

**Issue by NibblyPig** [qwerty-fr/qwerty-fr#42]
I use Windows 10 and have a UK layout. 

Qwerty-fr uses a US keyboard layout. There are some differences between the two, namely " and @ are reversed and some other symbols are moved around.

While the keyboard mostly works OK the backslash key is now < even though my < key still works, and it's not possible to remap this using a third party tool because the tool sees both keypresses as the same despite being different physical keys.

Would be nice if this worked with other language keyboards, or allowed remapping to fix the differences.

**Comment by devnoname120** [comment]
@NibblyPig Which OS do you use?

**Comment by NibblyPig** [comment]
Windows 10

**Comment by devnoname120** [comment]
> Windows 10
> 
> Title might not be correct, ISO is the keyboard I have, it's a UK vs US layout thing I think rather than ISO vs ANSI
> 
> As " and @ are reversed between UK and US layouts

Thanks! Could you edit the title back

**Comment by NibblyPig** [comment]
Thanks, have clarified the issues in the title & message body

**Comment by patadune** [comment]
FWIW, I have a fork based on a UK layout at https://github.com/patadune/qwerty-fr/tree/uk_qwerty-fr

I only have the linux variant pushed right now, but will also update macOS/Windows ones

I'm not sure if/how that could be integrated to the qwerty-fr project

**Comment by devnoname120** [comment]
@patadune Does my current qwerty-fr configuration work with UK keyboards? i.e. transforms them into US keyboards without any issues.

**Comment by patadune** [comment]
@devnoname120 yes, I mean, apart from the " and @ keys + some others being reversed, as @NibblyPig mentioned

This brings another question: Is qwerty-fr made with ANSI or ISO keyboards in mind? The screenshots shows ISO keyboards, but is there such a thing as retail ISO US QWERTY keyboards?

I think you could consider having two variants:

- ANSI US qwerty-fr (current implementation but with updated geometry/screenshots)
- ISO UK qwerty-fr (my fork)

**Comment by devnoname120** [comment]
I'm confused: I looked on Apple UK's store to see what the keyboard looks like and it indeed seems to match the current layout (minor the paragraph and the backtick keys), but that's an Apple peculiarity).

![image](https://github.com/qwerty-fr/qwerty-fr/assets/2824100/62c9df59-6ab1-4483-8483-acb88a1df790)

Source: https://www.apple.com/uk/shop/product/MMMR3B/A/magic-keyboard-with-touch-id-and-numeric-keypad-for-mac-models-with-apple-silicon-british-english-black-keys

`QWERTY-fr` isn't meant to be based on US International, but a strict superset of QWERTY. The philosophy of the keyboard is adding new characters and not move/remove any characters that are already there in the base layout.

I would like to avoid having any variants whatsoever unless it's absolutely required (so far it hasn't been). Maintaining variants is extra work, confuses users as to which they should choose, and will make it way harder to eventually integrate the layout into operating systems as built-in.

What do you think?

**Comment by NibblyPig** [comment]
Wikipedia has this article on layouts: https://en.wikipedia.org/wiki/British_and_American_keyboards

I can't speak for Apple, but that layout would be hard to type on because " and @ are reversed from what I would expect. It does have the £ sign though so it does seem to be UK in some respects.

**Comment by gdesmott** [comment]
> This brings another question: Is qwerty-fr made with ANSI or ISO keyboards in mind?

I got confused by this as well.  The screenshot shows an ISO design but with US variant, not an UK one, which is kinda strange.

**Comment by patadune** [comment]
> I got confused by this as well. The screenshot shows an ISO design but with US variant, not an UK one, which is kinda strange.

My guess is that qwerty-fr comes from an AZERTY layout that was "QWERTYfied" from an ANSI US variant. That would explain the less/greater signs remaining next to the left Shift key, as there is no such key on an ANSI keyboard

It would make more sense to start from existing QWERTY layouts and augment them with French keys

**Comment by gdesmott** [comment]
> What do you think?

I agree that it would be best to not have any variant but I'm afraid this would mean leaving out owners of ISO UK keyboards.

I've been using those since decades as having an ISO format is the only option to be able to use the `fr_FR` layout on a qwerty keyboard. With an ANSI you can type the `<` and  `>` keys which are essential for programmers.

I think it's very likely I'm not the only one with such keyboards and lots of potential `qwerty-fr` users will be left out without an UK variant.

**Comment by lcoulet** [comment]
We are many French in my company to use ISO QWERTY Keyboards while struggling to use French characters...  Would love to have such a driver to use. 
Moreover from some resellers in France (like Dell) as far as I can tell it's easier to get ISO UK QWERTY keyboards than ANSI US.  There's a real need there IMHO.

Could this driver be adapted easily to such layouts ? 

[ISO keyboard (105)](https://upload.wikimedia.org/wikipedia/commons/9/9c/ISO_keyboard_%28105%29_QWERTY_UK.svg)

**Comment by devnoname120** [comment]
Leaving this here so that I don't lose it:
https://www.farah.cl/Keyboardery/A-Visual-Comparison-of-Different-National-Layouts/#enUS

**Comment by AdrienLemaire** [comment]
I have a Pebble Keys 2 k380s (japanese qwerty keyboard), and would also like to have a qwerty-fr variant mapped to it.

![](https://upload.wikimedia.org/wikipedia/commons/thumb/b/bc/KB_Japanese.svg/2560px-KB_Japanese.svg.png)

The symbols |[]@:"* etc are all mapped differently, and I use them a lot :rofl: 

---

I was able to use @mark-gerarts's technique (https://github.com/qwerty-fr/qwerty-fr/issues/49#issuecomment-1405254634) to get the qwerty-fr layout on my xps 13 9320 keyboard using NixOS.

To get the layout on my bluetooth keyboard as well, I tried to duplicate it to have the `jp_qwerty-fr` layout available next to `us_qwerty-fr`, but xkb doesn't recognize it. 
Do you know what I have missed ?

rules/evdev.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE xkbConfigRegistry SYSTEM "xkb.dtd">
<xkbConfigRegistry version="1.1">
  <layoutList>
    <layout>
      <configItem>
        <name>us_qwerty-fr</name>
        <shortDescription>English (US, qwerty-fr)</shortDescription>
        <description>English (US, qwerty-fr)</description>
      </configItem>
    <layout>
    </layout>
      <configItem>
        <name>jp_qwerty-fr</name>
        <shortDescription>Japanese (JP, qwerty-fr)</shortDescription>
        <description>Japanese (JP, qwerty-fr)</description>
      </configItem>
    </layout>
  </layoutList>
</xkbConfigRegistry>
```

rules/evdev.lst
```lst
! layout
  us_qwerty-fr	English (US, qwerty-fr)
  jp_qwerty-fr	Japanese (JP, qwerty-fr)
```

```diff
~/.config/xkb on ☁️   ➜ diff symbols/us_qwerty-fr symbols/jp_qwerty-fr 
5c5
<     include "us(basic)"
---
>     include "jp(basic)"
8c8
<     name[Group1]= "US keyboard with french symbols - AltGr combination";
---
>     name[Group1]= "JP keyboard with french symbols - AltGr combination";
```

hypr/userprefs.conf
```conf
input {
    kb_layout = us_qwerty-fr
    resolve_binds_by_sym = 1
}
device {
  name = pebble-k380s
  kb_layout = jp_qwerty-fr
}
```
![image](https://github.com/user-attachments/assets/e71564ea-ce63-4e8e-9ddb-872a9c6286ee)

**Comment by devnoname120** [comment]
@AdrienLemaire Could you create a new discussion either in the qwerty-fr repo or on the hypr repo?

**Comment by devnoname120** [comment]
After careful consideration, I've decided not to implement a new version of the qwerty-fr based on the [`English (UK) ISO (standard)`](https://www.farah.cl/Keyboardery/A-Visual-Comparison-of-Different-National-Layouts/#enUK) and I decided to keep the *status quo*, which is that qwerty-fr is based on [`US ANSI/ISO (terminal style)`](https://www.farah.cl/Keyboardery/A-Visual-Comparison-of-Different-National-Layouts/#enUS).

The main reasons are:
- UK ANSI doesn't exist while US comes in both ANSI and ISO. This would make it very awkward for ISO users — which are the overwhelming majority of French users and European Union users — to base the qwerty-fr on a layout that doesn't have ISO at all.
- The UK ANSI already defines some <kbd>AltGr</kbd> keys that completely clash with the logic of the qwerty-fr. We would either need to unbind them completely (which would make the keyboard *not* a superset anymore), or change the qwerty-fr around them — but then placing letters such as “ù”, “ì”, and “ò” in a way that makes sense would be impossible. You could neither put the acute accents on the letters themselves (they are already occupied), nor on the left/right of the characters (it would clash with the already-used characters).
- Likewise, the placement of special characters in the UK is different from their placement on the US keyboard. That would mean a complete redesign of the layout to keep a logic that makes sense.
- The US ANSI/ISO layouts are the *lingua franca* of the keyboard world. One of the perks of the qwerty-fr is that it's based on the most popular keyboard layout on earth:
    - It is the default layout on pretty much every operating system, so it makes it an invaluable “fallback” layout for typing while RDP-ing, VNC-ing, KVM-ing, or SSH-ing into anything.
    - Almost all keyboard shortcuts and programming languages are designed with the US ANSI/ISO layout in mind rather than the UK ISO. Default shortcuts are defined to make them convenient to use on US ANSI/ISO layouts, programming languages use symbols that are easy to access on the US ANSI/ISO layouts, etc.
- I would rather give the $ (the #1 international trade currency) a privileged place rather than the sterling pound £ — a less important currency that divides rather than unifies, and that will hopefully disappear one day, replaced by the €).
- UK Mac keyboards are based on the US keyboard layout, not the Windows/Linux UK keyboard. So it wouldn't even be a good solution for 22% of users (the market share of Mac in the UK).
- I don't want to maintain both a US-based variant and a UK-based variant.
    - The goal of the qwerty-fr is to unify everything and having several variants would detract from that goal.
    - Furthermore having several variants would make it harder to get it to be eventually included by default on Windows, Linux, macOS, etc. or get it standardized at a standardization body, which is an eventual goal once the layout becomes stable/final.
    - It's also more maintenance and figuring out smart and intuitive ways to place characters on the US layout is already hard enough as it is. Maintaining two variants with different rules and logic would be a nightmare for me.

**Comment by mpsq** [comment]
> I would rather give the $ (the #1 international trade currency) a privileged place rather than the sterling pound £ — a less important currency that divides rather than unifies, and that will hopefully disappear one day, replaced by the €.

Well, the British pound is the oldest currency that is still in use today, which dates back to around 775 AD. I wouldn't bet on it disappearing anytime soon.

**Comment by devnoname120** [comment]
@mpsq Can you change the `KBD	qwertyfr	"French qwerty keyboard"` and in `DESCRIPTIONS` the `0409	French qwerty keyboard`? It will avoid conflicts.

Make sure to rebuild the `qwertyuk-fr_0.1.0_windows.tar.gz` afterwards. Thanks
